.github/workflows/gh-build-and-test.yml

```diff
@@ -1,16 +1,16 @@
 on:
   workflow_call:
     inputs:
-      host-platform:
```

Comment:
This could be saved for the next PR, but having this called build-and-test has been misleading imo. I think it could be better to call gh-build.yml and gh-test.yml from ci-gh.yml

Reply:
Discussed on the call, yes we'll have to refactor this to accommodate that we have different build/test matrices to cover.